src/compiler/builderState.ts

```diff
@@ -201,7 +261,7 @@ namespace ts {
         /**
          * Returns true if oldState is reusable, that is the emitKind = module/non module has not changed
          */
-        export function canReuseOldState(newReferencedMap: ReadonlyESMap<Path, ReferencedSet> | undefined, oldState: Readonly<ReusableBuilderState> | undefined) {
+        export function canReuseOldState(newReferencedMap: TwoWayMap<Path, Path> | undefined, oldState: Readonly<ReusableBuilderState> | undefined) {

```

Comment:
Can you also defined ReadOnlyTwoWayMap which doesnt allow delete or changes to entries,
I had used readonly here to ensure we are not changing oldState.referenceMap or oldState.exportedMaps anywhere.

Reply:
Sure.